app/coffee/DocArchiveManager.coffee

```diff
@@ -26,21 +27,26 @@ module.exports = DocArchive =
 	archiveDoc: (project_id, doc, callback)->
 		logger.log project_id: project_id, doc_id: doc._id, "sending doc to s3"
 		try
-			options = DocArchive.buildS3Options(doc.lines, project_id+"/"+doc._id)
+			options = DocArchive.buildS3Options(project_id+"/"+doc._id)
 		catch e
 			return callback e
-		request.put options, (err, res)->
-			if err? || res.statusCode != 200
-				logger.err err:err, res:res, project_id:project_id, doc_id: doc._id, statusCode: res?.statusCode, "something went wrong archiving doc in aws"
-				return callback new Error("Error in S3 request")
-			md5lines = crypto.createHash("md5").update(JSON.stringify(doc.lines), "utf8").digest("hex")
-			md5response = res.headers.etag.toString().replace(/\"/g, '')
-			if md5lines != md5response
-				logger.err responseMD5:md5response, linesMD5:md5lines,  project_id:project_id, doc_id: doc?._id, "err in response md5 from s3"
-				return callback new Error("Error in S3 md5 response")
-			MongoManager.markDocAsArchived doc._id, doc.rev, (err) ->
-				return callback(err) if err?
-				callback()
+		DocArchive._mongoDocToS3Doc doc, (error, json_doc) ->
+			return callback(error) if error?
+			options.body = json_doc
+			options.headers =
+				'Content-Type': "application/json"
```

Comment:
We previously set it via `json: doc`, but we need to do it manually now to have the JSON string to compare the md5 hash to.